client/my-sites/site-settings/site-tools/index.jsx

```diff
@@ -190,6 +205,9 @@ export default compose( [
 				sourceSlug: siteSlug,
 			} );
 
+			const showStartSiteTransfer =
+				isEnabled( 'yolo/site-transfers-i1' ) && ! isJetpack && ! isVip && ! isP2Hub && ! isAtomic;
```

Comment:
Should we check if the user is a site owner? Ideally, we could reuse a function like `canStartSiteOwnerTransfer` for this condition.

Now, non-owner admins that are not site owners can see the "CTA" but when the user clicks on it, is redirected to the same page - General Settings.

https://github.com/Automattic/wp-calypso/assets/779993/4ca9992c-c68a-44ac-b719-af5a1d39beff

## steps to reproduce
- Add a user as admin
- Accept the invitation
- With that non-owner admin go to Settings > General
- Click on `Transfer your site`